include/picolibrary/testing/interactive/gpio.h

```diff
@@ -30,6 +30,58 @@
  */
 namespace picolibrary::Testing::Interactive::GPIO {
 
+/**
+ * \brief GPIO input pin state interactive test helper.
+ *
+ * \tparam Output_Stream The type of asynchronous serial output stream to use to output
+ *         information to the user.
+ * \tparam Transmitter The type of asynchronous serial transmitter to use to transmit
+ *         information to the user.
+ * \tparam Input_Pin The type of input pin to toggle.
+ * \tparam Delayer A nullary functor called to introduce a delay each time the pin's state
+ *         is read
+ *
+ * \param[in] transmitter The asynchronous serial transmitter to use to transmit
+ *            information to the user.
+ * \param[in] pin The pin to toggle.
+ * \param[in] delay The nullary functor called to introduce a delay each time the pin's
+ *            state is read.
+ */
+template<template<typename> typename Output_Stream, typename Transmitter, typename Input_Pin, typename Delayer>
+void state( Transmitter transmitter, Input_Pin pin, Delayer delay ) noexcept
+{
+    // #lizard forgives the length
+
+    auto stream = Output_Stream{ std::move( transmitter ) };
+
+    if ( stream.initialize().is_error() ) {
+        return;
+    } // if
+
+    {
+        auto const result = pin.initialize();
+        if ( result.is_error() ) {
+            static_cast<void>( stream.print( "pin initialization error: {}\n", result.error() ) );
+
+            return;
+        } // if
+    }
+
+    for ( ;; ) {
+        delay();
+
+        auto const result = pin.state();
+        if ( result.is_error() ) {
+            static_cast<void>( stream.print( "pin state error: {}\n", result.error() ) );
+
+            return;
+        } // if
+        else {
+            static_cast<void>( result.is_high() ? stream.print( "pin is high\n" ) : stream.print( "pin is low\n" ) );
+        } // else
+    }     // for
```

Comment:
There are a few things going on here:
1. The error detected on line 74 is handled by the early return (which avoids any need for deeply nested if statements) on line 77. Therefore, the else clause (line 79) of the if statement is not actually needed. In cases where an else clause is needed, the closing bracket of the if clause would not be commented, and `else` would be placed on the same line as the if clause's closing bracket.
2. A GPIO input pin's `state()` function returns either `::picolibrary::Result<::picolibrary::GPIO::Pin_State, ::picolibrary::Error_Code>` or `::picolibrary::Result<::picolibrary::GPIO::Pin_State, ::picolibrary::Void>` (see https://github.com/apcountryman/picolibrary/blob/ceef8173e2521c4348e74454091cd0ffc8551859/include/picolibrary/gpio.h#L168-L177). The result's `value()` statement must be called to access the pin state.
3. The call(s) to `print()` on line 80 are part of the normal program flow. Therefore, the result of the call(s) to `print()` must be checked to see if the print succeeded. If the call(s) to `print()` fail, the function has encountered a fatal error and should return. The other calls to `print()` in this function occur when a fatal error has already been encountered and thus the result of the calls to `print()` do not need to be checked as there is nothing that can be done.
4. The use of a ternary on line 80 is unusual, though it does work. Using either a format string with a single call to `print()`, or using a ternary to select which string is passed to `put()` (avoiding format string parsing overhead) would be clearer.

Putting it all together (may not pass the formatting check):
```suggestion
        auto const result = pin.state();
        if ( result.is_error() ) {
            static_cast<void>( stream.print( "pin state getting error: {}\n", result.error() ) );

            return;
        } // if

        if ( stream.print( "pin is {}\n", result.value().is_high() ? "high" : "low" ).is_error()) {
            return;
        } // if
    }     // for
```